### PR TITLE
azure: Convert Windows installations scripts to `bash`

### LINKED ISSUE #64551: azure: CI doesn't fail if msys2 fails to install
The "Install MSYS2" step of [this build](https://dev.azure.com/rust-lang/rust/_build/results?buildId=8318&view=logs&j=4bc7ef66-8a59-503f-fa7a-b9533d6853d5) failed, but the build kept going and failed later with obscure errors. We should fix this so that if msys2 installation fails the build fails immediately!

### PR DESCRIPTION
**Pull request by alexcrichton**
Looks like `script`, which uses `cmd.exe`, doesn't have fail-fast
behavior and if a leading command fails the script doesn't actually fail
so long as the last command succeeds. We instead want the opposite
behavior where if any step fails the whole script fails.

I don't really know `cmd.exe` that well, nor powershell, so I've opted
to move everything to `bash` which should be a good common denominator
amongst all platforms to work with. Additionally I know that `set -e`
works to cause scripts to fail fast.

Closes #64551

### CODE REVIEW

File: src/ci/azure-pipelines/steps/install-windows-build-deps.yml [hunk 1]
```diff
@@ -18,9 +18,9 @@ steps:
 # one is MSI installers and one is EXE, but they're not used so frequently at
 # this point anyway so perhaps it's a wash!
 - script: |
-    powershell -Command "$ProgressPreference = 'SilentlyContinue'; iwr -outf is-install.exe https://rust-lang-ci-mirrors.s3-us-west-1.amazonaws.com/rustc/2017-08-22-is.exe"
-    is-install.exe /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-
     echo ##vso[task.prependpath]C:\Program Files (x86)\Inno Setup 5
+    curl.exe -o is-install.exe https://rust-lang-ci-mirrors.s3-us-west-1.amazonaws.com/rustc/2017-08-22-is.exe
+    is-install.exe /VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-
```
Comment (Mark-Simulacrum) [inline src/ci/azure-pipelines/steps/install-windows-build-deps.yml]: Hm, Is there a reason we're moving this after the `echo` statement? It looks like these two might be connected

**Review by Mark-Simulacrum — APPROVED** [review]

**Comment by Mark-Simulacrum** [comment]
Oh, actually, I forgot I left that comment towards the start of my review :)

r=me with an answer/fix

@bors r-

File: src/ci/azure-pipelines/steps/install-windows-build-deps.yml (hunk 1 above)
Comment (alexcrichton) [reply src/ci/azure-pipelines/steps/install-windows-build-deps.yml]: Ah yeah I couldn't figure out how to get this working with `bash` and AFAIK the exit code of the script is the last statement, so the thing which may fail is listed here last and the `echo`, which always succeeds, was moved above it.

**Pull request merged**